engine/src/main/java/io/zeebe/engine/processing/bpmn/ProcessInstanceStateTransitionGuard.java

```diff
@@ -59,8 +59,8 @@ public boolean isValidStateTransition(final BpmnElementContext context) {
     // processors in the processor. this means that ELEMENT_COMPLETING of a migrated processor will
     // set the element state to ELEMENT_COMPLETING on replay, but ELEMENT_COMPLETING of a non
     // migrated will expect its state to ALREADY be ELEMENT_COMPLETING. To avoid too much
-    // complexity, when reprocessing events for non migrated processors, just set their state as
-    // expected; this could in the future be a source of error, but as its only temporary until
+    // complexity, when reprocessing events for non migrated processors, just allow their state as
```

Comment:
Nit: very tiny nit, but I guess `just allow the transition` works better than `their state` :sweat_smile:

Reply:
applied